src/Metrics/DefaultMetricsHandler.php

```diff
@@ -31,22 +31,22 @@ public function handleMetrics(Feature $feature, bool $successful, Variant $varia
         }
     }
 
-    private function getOrCreateBucket(): MetricsBucket
+    private function getOrCreateBucket(): DefaultMetricsBucket
```

Comment:
This should stay unchanged, meaning the return type hint should now point to the interface.

```suggestion
    private function getOrCreateBucket(): MetricsBucket
```

Reply:
Then it would mean the `MetricsBucket` interface must expose an `addToggle` function, which takes a `MetricsBucketToggle` as an argument, which in turns need to be transformed in an interface... while the fact that the Bucket works with MetricsBucketToggle is an implementation detail IMHO.

It is also an implementation detail that the `DefaultMetricsHandler` works with `DefaultMetricsBucket`, and since that function is private, I don't think there's any benefit of returning the interface instead of the concrete class IMHO :)